### PR TITLE
Transaction Fees: Pay for failed transactions

### PR DESCRIPTION
**Pull request by janezpodhostnik**
closes: https://github.com/dapperlabs/flow-internal/issues/1449

- Payer still pays even if transaction fails
- if storage fails transaction invocation is reverted
- sequence number is still increased if transaction fails.
- Transaction fees are sent to the FlowFees contract ref: https://github.com/dapperlabs/flow-internal/issues/1481

## TODO

- [ ] If authoriser signature fails still deduct TX fees and increment sequence number

### CODE REVIEW

File: fvm/transaction.go [hunk 1]
```diff
@@ -71,11 +72,13 @@ func (proc *TransactionProcedure) Run(vm *VirtualMachine, ctx Context, st *state
 
 type TransactionInvocator struct {
 	logger zerolog.Logger
+	checkStorage bool
 }
 
-func NewTransactionInvocator(logger zerolog.Logger) *TransactionInvocator {
+func NewTransactionInvocator(logger zerolog.Logger, checkStorage bool) *TransactionInvocator {
```
Comment (ramtinms) [inline fvm/transaction.go:78]: I think we should keep the params like checkStorage as part of the context which are passed to the Process.

**Comment by ramtinms** [comment]
@janezpodhostnik is this still a relevant PR? should I review it again?

**Comment by janezpodhostnik** [comment]
@ramtinms no, this is partially already included in master. And I have to redo the rest. I changed it to a draft.

**Comment by janezpodhostnik** [comment]
A lot has changed since this PR. I will create a new one.

**Pull request closed without merging**